frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -14,28 +14,124 @@
  * limitations under the License.
  */
 
-import { createChart, IChartApi, LineData } from 'lightweight-charts';
+import { createChart, IChartApi, LineData, LineStyle, ISeriesApi, UTCTimestamp } from 'lightweight-charts';
 import { Component, createRef, h, RenderableProps } from 'preact';
 import { translate, TranslateProps } from '../../../decorators/translate';
+import { formatAsCurrency, Fiat } from '../../../components/rates/rates';
+import * as styles from './chart.css';
 
 export type ChartData = LineData[];
 
 interface ChartProps {
     // If undefined, data is missing.
-    data?: ChartData;
+    dataDaily?: ChartData;
+    dataHourly?: ChartData;
+    fiatUnit: Fiat;
 }
 
 interface State {
+    display: 'week' | 'month' | 'year' | 'all';
+    source: 'daily' | 'hourly';
+    currentTotal?: number;
+    difference?: number;
+    diffSince?: string;
 }
 
 type Props = ChartProps & TranslateProps;
 
 class Chart extends Component<Props, State> {
     private ref = createRef();
     private chart?: IChartApi;
+    private lineSeries?: ISeriesApi<'Area'>;
     private resizeTimerID?: any;
     private height: number = 300;
 
+    public readonly state: State = {
+        display: 'all',
+        source: 'daily',
+        currentTotal: this.props.dataDaily && this.props.dataDaily[this.props.dataDaily.length - 1].value,
```

Comment:
this should not be a state var, as it never changes. you can make a local function for example: `private getCurrentTotal()` to calculate it.

Reply:
also: use the hourly dataset instead. is better for currentTotal, as it goes to the current hour. the daily dataset is at utc midnight of the last day, so missing a few hours.